src/git-octopus

```diff
@@ -73,46 +82,69 @@ fi
 
 branches=$(git ls-remote . $patterns | cut -d $'\t' -f 2)
 
+[ $branches -n ] && die "No branch matching $patterns were found"
```

Comment:
I agree that this could be a normal usecase, maybe just a warning and exit 0 is better